### PR TITLE
Change format of create form

### PR DESCRIPTION
**Pull request by Ozrei**
…another key

### CODE REVIEW

File: keyform/models.py [hunk 1]
```diff
@@ -46,31 +46,24 @@ class Request(models.Model):
...
-        verbose_name=_('Amount received'),
-        validators=[
-            MinValueValidator(
-                Decimal('0.00'))])
-    payment_method = models.CharField(max_length=2, choices=PAYMENT_TYPES, null=True, blank=True)
-    charge_amount = models.DecimalField(
-        max_digits=7, decimal_places=2, validators=[
-            MinValueValidator(
-                Decimal('0.00'))])
+    STATUS_TYPES = (
```
Comment (shaneallgeier) [inline keyform/models.py]: This is no longer used.

File: keyform/static/keyform/css/add_form.css [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
-	color: red;
+div.row {
+    padding: 15px;
 }
 
-td {
-	text-align: left;
+.errorlist {
+    color: red;
 }
```
Comment (shaneallgeier) [inline keyform/static/keyform/css/add_form.css]:
```css
.keydata-form .generate_forms {
    visibility: hidden;
}

.keydata-form:last-of-type .generate_forms {
    visibility: visible;
}
```

File: keyform/templates/keyform/add_form.html [hunk 3]
```diff
@@ -14,97 +14,69 @@
...
+            </div>
+            <div class="large-12 columns">
+                {{ form.student_name.errors }}
+                {{ form.student_name.label_tag }}
+                <span class="help-text">{{ form.student_name.help_text }}</span>
+                {{ form.student_name }}
+            </div>
+            <div id="form-set">
+                {% for keydata_form in form.request_formset.forms %}
+                    {% include 'keyform/includes/keydata_form.html' with include_buttons=True %}
```
Comment (shaneallgeier) [inline keyform/templates/keyform/add_form.html]: `include_buttons` no longer exists in the included template, so it's no longer necessary here.

File: keyform/forms.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class Meta:
     def __init__(self, *args, **kwargs):
         super(CreateForm, self).__init__(*args, **kwargs)
         # removes blank choices from Radio Select options
-        self.fields['payment_method'] = TypedChoiceField(widget=RadioSelect(), choices=Request.PAYMENT_TYPES)
+        self.fields['payment_method'] = TypedChoiceField(widget=RadioSelect(), choices=[('', _('Not Applicable'))] + list(Request.PAYMENT_TYPES), required=False)
```
Comment (aaaustin10) [inline keyform/forms.py]: Why wasn't 'Not Applicable' added to the models instead of being added here?
I don't see why required is set to false here either.

Comment (shaneallgeier) [inline keyform/forms.py]: Because I wanted it to store as NULL for some reason... I don't remember that reason, though...

If you really want to change it, I'm open to that

Comment (aaaustin10) [inline keyform/forms.py]: If I'm not mistaken that will just store them as the empty string.

**Comment by shaneallgeier** [comment]
I'm pretty sure it stored NULL/None when I tested it.
Test it.

**Comment by shaneallgeier** [comment]
Well that's embarrassing. Change it I guess

File: keyform/forms.py [hunk 5]
```diff
@@ -2,10 +2,10 @@
 from django.forms.widgets import RadioSelect, CheckboxSelectMultiple
 from django.forms import TypedChoiceField
 from django.forms.models import inlineformset_factory
+from django.utils.translation import ugettext_lazy as _
```
Comment (enki211) [inline keyform/forms.py]: Is this needed?

File: keyform/models.py [hunk 6]
```diff
@@ -44,33 +44,21 @@ class Request(models.Model):
...
-        verbose_name=_('Amount received'),
-        validators=[
-            MinValueValidator(
-                Decimal('0.00'))])
-    payment_method = models.CharField(max_length=2, choices=PAYMENT_TYPES, null=True, blank=True)
-    charge_amount = models.DecimalField(
-        max_digits=7, decimal_places=2, validators=[
-            MinValueValidator(
-                Decimal('0.00'))])
+    bpn_validator = RegexValidator('[mM8]\d{8}', "Bearpass number must start with an 'M,' 'm,' or '8,' and followed by eight digits.")
```
Comment (enki211) [inline keyform/models.py]: Should be translated

File: keyform/models.py [hunk 7]
```diff
@@ -44,33 +44,21 @@ class Request(models.Model):
...
-                Decimal('0.00'))])
-    staff = models.ForeignKey(settings.AUTH_USER_MODEL)
-    bpn = models.CharField(max_length=9, verbose_name=_('M-Number'), validators=[bpn_validator])
+    bpn_validator = RegexValidator('[mM8]\d{8}', _("Bearpass number must start with an 'M,' 'm,' or '8,' and followed by eight digits."))
+
+    building = models.ForeignKey(Building, help_text='')
+    student_name = models.CharField(max_length=128, blank=True, help_text='')
+    reason_for_request = models.CharField(max_length=2, choices=REQUEST_TYPES, default='dk', help_text='')
+    amt_received = models.DecimalField(max_digits=7, decimal_places=2, default=0, blank=True, verbose_name= _('Amount received'), validators=[MinValueValidator(Decimal('0.00'))], help_text='')
+    payment_method = models.CharField(max_length=2, choices=PAYMENT_TYPES, null=True, blank=True, default='ca', help_text='')
```
Comment (shaneallgeier) [inline keyform/models.py]: Since we have the new "Not Applicable" option, it no longer has the need to be nullable/blankable. Also I think we should remove the default just because there's no reason for it to be selected on page load.

File: keyform/models.py [hunk 8]
```diff
@@ -47,15 +47,15 @@ class Request(models.Model):
         ('na', _('Not Applicable')),
     )
 
-    bpn_validator = RegexValidator('[mM8]\d{8}', "Bearpass number must start with an 'M,' 'm,' or '8,' and followed by eight digits.")
+    bpn_validator = RegexValidator('[mM8]\d{8}', _("Bearpass number must start with an 'M,' 'm,' or '8,' and followed by eight digits."))
 
     building = models.ForeignKey(Building, help_text='')
     student_name = models.CharField(max_length=128, blank=True, help_text='')
-    reason_for_request = models.CharField(max_length=2, choices=REQUEST_TYPES, help_text='')
+    reason_for_request = models.CharField(max_length=2, choices=REQUEST_TYPES, default='dk', help_text='')
```
Comment (shaneallgeier) [inline keyform/models.py]: Why did you add a default to this?

Comment (Ozrei) [inline keyform/models.py]: Commenting for potential future face-rubbing xx

File: keyform/forms.py [hunk 9]
```diff
@@ -17,6 +18,38 @@ def __init__(self, *args, **kwargs):
...
+        amt_received = cleaned_data.get("amt_received")
+        payment_method = cleaned_data.get("payment_method")
+        bpn = cleaned_data.get("bpn")
+        student_name = cleaned_data.get("student_name")
+        charge_amount = cleaned_data.get("charge_amount")
+        charged_on_rcr = cleaned_data.get("charged_on_rcr")
+
+        if reason_for_request == "lk":
+            #require new core number when staff key or lost key
+            if bpn == "":
```
Comment (enki211) [inline keyform/forms.py]: Instead of checking for the empty string what about 
```python
if not bpn:

File: keyform/forms.py [hunk 10]
```diff
@@ -17,6 +18,38 @@ def __init__(self, *args, **kwargs):
...
+                error_msg = _("Must have Bearpass Number when Lost/Stolen Key.")
+                self.add_error('bpn', error_msg)
+
+            if student_name == "":
```
Comment (enki211) [inline keyform/forms.py]: Here as well

File: keyform/forms.py [hunk 11]
```diff
@@ -36,4 +69,20 @@ class Meta:
         fields = ['status']
 
 
+class KeyDataForm(forms.ModelForm):
```
Comment (enki211) [inline keyform/forms.py]: It does not look like this is being used anywhere.

File: keyform/forms.py [hunk 12]
```diff
@@ -18,6 +18,38 @@ def __init__(self, *args, **kwargs):
...
+                error_msg = _("You must pick a billing method.")
+                self.add_error(None, error_msg)
+                error_msg = _("Choose one.")
+                self.add_error('amt_received', error_msg)
+                self.add_error('charge_amount', error_msg)
+                self.add_error('charged_on_rcr', error_msg)
+
+        if amt_received > 0 and payment_method == "na":
+            error_msg = _("If Amount Received is greater than zero, Payment Method must be selected.")
+            self.add_error('payment_method', error_msg)
```
Comment (shaneallgeier) [inline keyform/forms.py]: You still need to `return cleaned_data` at the end

File: keyform/forms.py [hunk 13]
```diff
@@ -18,6 +18,38 @@ def __init__(self, *args, **kwargs):
...
+        reason_for_request = cleaned_data.get("reason_for_request")
+        amt_received = cleaned_data.get("amt_received")
+        payment_method = cleaned_data.get("payment_method")
+        bpn = cleaned_data.get("bpn")
+        student_name = cleaned_data.get("student_name")
+        charge_amount = cleaned_data.get("charge_amount")
+        charged_on_rcr = cleaned_data.get("charged_on_rcr")
+
+        if reason_for_request == "lk":
+            #require new core number when staff key or lost key
```
Comment (shaneallgeier) [inline keyform/forms.py]: This comment doesn't make any sense in this context

File: keyform/forms.py (hunk 12 above)
Comment (shaneallgeier) [inline keyform/forms.py]: We also need to show an error on `amt_received` if `amt_received == 0 and payment_method != "na"`
